### PR TITLE
Automate release process

### PR DESCRIPTION
**Pull request by machadovilaca**
This PR adds an automatic release procedure of the must-gather image. When a new tag is pushed, it creates a new
GitHub release for that tag with notable changes since the last release and builds and creates
an image, and publishes it in quay.io.

An example release from this tool execution can be found at
https://github.com/machadovilaca/must-gather/releases/tag/v1.0.0.

Before this tool can be used, a previous release must already exist. Since the tag
https://github.com/kubevirt/must-gather/releases/tag/v0.1.0 already exists, we just need to create a release for this
tag, such as in https://github.com/machadovilaca/must-gather/releases/tag/v0.1.0.
`QUAY_USERNAME` and `QUAY_PASSWORD` GitHub Actions secrets must also be created, which access to KubeVirt quay.

Signed-off-by: João Vilaça <jvilaca@redhat.com>

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: scripts/release/release.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	flag.Set("stderrthreshold", "WARNING")
+	flag.Set("v", "2")
+
+	version := flag.String("version", "", "Release version tag. Must be a valid semver. The branch is automatically detected from the major and minor release")
+	githubTokenFile := flag.String("github-token-file", "", "File containing the github token.")
+	quayUsername := flag.String("quay-username", "", "Quay username.")
+	quayTokenFile := flag.String("quay-token-file", "", "File containing the quay token.")
+	flag.Parse()
+
+	if *githubTokenFile == "" {
```
Comment (nunnatsa) [inline scripts/release/release.go:34]: What is the parameter is missing? nil or pointer to an empty string?

**Review by nunnatsa — COMMENTED** [review]
I'm not sure about this PR. 

I wander if everything should be done in golang. How easy is debugging the process if it fails?

I would like  to get more opinions. @tiraboschi @orenc1 - WDYT?

File: scripts/release/release.go (hunk 1 above)
Comment (machadovilaca) [reply scripts/release/release.go:34]: it's an empty string because of the default value given by the `flag.String` function

**Comment by tiraboschi** [comment]
> I would like to get more opinions. @tiraboschi @orenc1 - WDYT?

Yes, I also feel this a bit overkilling.
Probably a couple of bash scripts executed by a github actions (with full logs line by line) are more maintainable.
I'd ideally use a go tool only to generate the release note.

**Comment by machadovilaca** [comment]
mhhh, okay
wdyt about a github action triggered by creating a tag through terminal/browser, and then the action would:
 - use the go tool to generate release notes
 - create github release
 - build and push image to quay
 
 sounds good @nunnatsa @tiraboschi ?

**Review by nunnatsa — APPROVED** [review]
/approve

**Pull request merged**